File: backend/src/main/java/org/jdk/project/service/SignService.java
package org.jdk.project.service;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.jdk.project.dto.sign.SignInDto;
import org.jdk.project.dto.sign.SignUpDto;
import org.jdk.project.exception.BusinessException;
import org.jdk.project.repository.UserRepository;
import org.jooq.generated.project.tables.pojos.User;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

/** 登录注册相关业务逻辑。 */
@Service
@Slf4j
@RequiredArgsConstructor
public class SignService {

  private final UserRepository userRepository;

  /**
   * 用户登录。
   *
   * @param signInDto 登录请求（用户名/密码）
   * @return 登录成功后的用户ID
   */
  public Long signIn(SignInDto signInDto) {
    User user = userRepository.fetchEnabledUserByUsername(signInDto.getUsername());
    if (user == null) {
      throw new BusinessException("用户名或密码错误");
    }
    if (!signInDto.getPassword().equals(user.getPassword())) {
      throw new BusinessException("用户名或密码错误");
    }
    return user.getId();
  }

  /**
   * 用户注册，并绑定默认角色。
   *
   * @param signUpDto 注册请求（用户名/密码）
   */
  @Transactional(rollbackFor = Throwable.class)
  public void signUp(SignUpDto signUpDto) {
    // 用户名唯一性校验
    if (userRepository.fetchOneByUsername(signUpDto.getUsername()) != null) {
      throw new BusinessException("用户名已存在");
    }
    User user = new User();
    user.setUsername(signUpDto.getUsername());
    user.setPassword(signUpDto.getPassword());
    user.setEnable(true);
    userRepository.insert(user);
  }
}

File: backend/src/main/kotlin/org/jdk/project/service/SignService.kt
package org.jdk.project.service

import org.jdk.project.dto.sign.AuthUserDto
import org.jdk.project.dto.sign.SignInDto
import org.jdk.project.dto.sign.SignUpDto
import org.jdk.project.exception.BusinessException
import org.jdk.project.repository.UserRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class SignService(
    private val userRepository: UserRepository,
) {
    fun signIn(signInDto: SignInDto): Long? {
        val user =
            userRepository.findEnabledUserByUsername(signInDto.username)
                ?: throw BusinessException("用户名或密码错误")
        if (signInDto.password != user.password) {
            throw BusinessException("用户名或密码错误")
        }
        return user.id
    }

    @Transactional(rollbackFor = [Throwable::class])
    fun signUp(signUpDto: SignUpDto) {
        if (userRepository.findUserByUsername(signUpDto.username) != null) {
            throw BusinessException("用户名已存在")
        }
        userRepository.insertCredentials(signUpDto.username, signUpDto.password)
    }

    fun getCurrentUser(username: String): AuthUserDto? =
        userRepository.findUserByUsername(username)?.let { user ->
            AuthUserDto(
                id = user.id,
                username = user.username,
                nickname = user.nickname,
                email = user.email,
                phone = user.phone,
                sex = user.sex,
                avatarUrl = user.avatarUrl,
                createTime = user.createTime,
                updateTime = user.updateTime,
                password = null,
                enable = user.enable,
                remark = user.remark,
            )
        }
}
